client/state/ui/guided-tours/contexts.js

```diff
@@ -19,15 +20,20 @@ export const isPreviewNotShowing = state =>
 export const isPreviewShowing = state =>
 	isPreviewShowingSelector( state );
 
+const timeSinceUserRegistration = state => {
+	const user = getCurrentUser( state );
+	return user ? ( Date.now() - Date.parse( user.date ) ) : false;
+};
+
 const WEEK_IN_MILLISECONDS = 7 * 1000 * 3600 * 24;
 export const isNewUser = state => {
-	const user = getCurrentUser( state );
-	if ( ! user ) {
-		return false;
-	}
+	const userAge = timeSinceUserRegistration( state );
+	return ! isNaN( userAge ) ? userAge <= WEEK_IN_MILLISECONDS : false;
+};
 
-	const creation = Date.parse( user.date );
-	return ( Date.now() - creation ) <= WEEK_IN_MILLISECONDS;
+export const isUserOlderThan = age => state => {
+	const userAge = timeSinceUserRegistration( state );
+	return ! isNaN( userAge ) ? userAge >= age : false;
```

Comment:
this exact code was actually pushed to this branch: `userAge ? userAge >= age : false`

The problem was it failed our tests, as the difference between user registration was 0 and that failed the naïve condition. I replaced NaN magic with explicit `userAge !== false`, which should be clear